### PR TITLE
Fix #549: Use server default port when port is not specified in conne…

### LINKED ISSUE #549: Make port in connection URI optional
It seems that even though we have on the server side a concept of default port - `9510` for now - this is not something we can rely on when opening a connection.
The following connection URI causes and `AssertionError`: `terracotta://localhost` even when a server on `9510` is started on the machine.

Could support for making the port optional in the connection URI be added?

### PR DESCRIPTION
**Pull request by ramsai1729**
…ction URI

### CODE REVIEW

File: dso-l1/src/main/java/com/tc/config/schema/L2ConfigForL1.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public L2Data(String host, int tsaPort) {
     public L2Data(String host, int tsaPort, boolean secure) {
       Assert.assertNotBlank(host);
       this.host = host;
-      this.tsaPort = tsaPort;
+      this.tsaPort = tsaPort < 0 ? 9510 : tsaPort; // use default port if given one is not valid
       this.secure = secure;
```
Comment (myronkscott) [inline dso-l1/src/main/java/com/tc/config/schema/L2ConfigForL1.java]: Shouldn't you use the new default port here?

Comment (ramsai1729) [inline dso-l1/src/main/java/com/tc/config/schema/L2ConfigForL1.java]: its the correct port now, can be changed after new tc-config release.

**Comment by myronkscott** [comment]
please rebase

**Comment by ramsai1729** [comment]
rebased

**Pull request merged**